resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Constants.scala

```diff
@@ -54,6 +54,7 @@ private[spark] object Constants {
   val UI_PORT_NAME = "spark-ui"
 
   // Environment Variables
+  val ENV_DRIVER_POD_IP = "SPARK_DRIVER_POD_IP"
```

Comment:
Is it different to `DRIVER_HOST_ADDRESS`? I saw K8s uses `DRIVER_HOST_ADDRESS` to derive driver url for env var `ENV_DRIVER_URL`.

Reply:
Thank you for review, @viirya . 

In K8s, `DRIVER_HOST_ADDRESS` is protected by `DriverServiceFeatureStep` here.

https://github.com/apache/spark/blob/7fccd676a27540b88fbcf951e53b6e43fdc26562/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/DriverServiceFeatureStep.scala#L109

https://github.com/apache/spark/blob/7fccd676a27540b88fbcf951e53b6e43fdc26562/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/DriverServiceFeatureStep.scala#L38-L40

It's because we inject like this systematically.

https://github.com/apache/spark/blob/7fccd676a27540b88fbcf951e53b6e43fdc26562/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/DriverServiceFeatureStep.scala#L67-L68

However, when DNS doesn't work, we need IP which is unknown from the executor pods so far.